### PR TITLE
[EP-2410] Implement branded checkout datalayer

### PR DESCRIPTION
**Pull request by canac**
Implement the branded checkout datalayer spec.

This PR also fixes a bug with the add payment modal being stuck in a loading state in branded checkout.

https://jira.cru.org/projects/browse/EP-2410

### CODE REVIEW

**Comment by canac** [comment]
Did basic testing in staging ✅

File: src/app/branded/analytics/branded-analytics.factory.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            quantity: '1'
+          }]
+        }
+      })
+    }),
+
+    // saveDonorDetails and savePaymentType should have been called before this
+    addPaymentInfo: suppressErrors(function (itemConfig) {
+      $window.dataLayer = $window.dataLayer || []
+      $window.dataLayer.push({ ecommerce: null })
```
[the remark this answers is not in the document]
Comment (canac) [reply src/app/branded/analytics/branded-analytics.factory.js:73]: Honestly, I'm not sure. It was in all of the spec slides, so I was just trusting that it's there for a reason.

File: src/app/analytics/analytics.factory.js [hunk 2]
```diff
@@ -9,6 +9,27 @@ import isEmpty from 'lodash/isEmpty'
 
 const analyticsFactory = /* @ngInject */ function ($window, $timeout, sessionService) {
   return {
+    fieldError: function (field, error) {
```
[the remark this answers is not in the document]
Comment (canac) [reply src/app/analytics/analytics.factory.js]: Renamed to `checkoutFieldError` and added checks to ensure that these events are only sent during a checkout flow.

File: src/app/checkout/step-2/step-2.component.js [hunk 3]
```diff
@@ -50,7 +52,9 @@ class Step2Controller {
   }
 
   handlePaymentChange (selectedPaymentMethod) {
-    this.defaultPaymentType = selectedPaymentMethod['account-type'] ? selectedPaymentMethod['account-type'] : selectedPaymentMethod['card-type']
+    const paymentType = selectedPaymentMethod['account-type'] ? selectedPaymentMethod['account-type'] : selectedPaymentMethod['card-type']
```
Comment (dr-bizz) [inline src/app/checkout/step-2/step-2.component.js:55]: I'm unsure of the JS stadards in this document, but a simplified way to do this is the below.
```suggestion
const paymentType = selectedPaymentMethod['account-type'] ||  selectedPaymentMethod['card-type']

File: src/app/branded/step-2/branded-checkout-step-2.spec.js [hunk 4]
```diff
@@ -21,20 +21,28 @@ describe('branded checkout step 2', () => {
...
     it('should load cart data', () => {
-      jest.spyOn($ctrl.cartService, 'get').mockReturnValue(Observable.of('some data'))
+      const cartData = { items: [] }
+      jest.spyOn($ctrl.cartService, 'get').mockReturnValue(Observable.of(cartData))
       $ctrl.loadCart()
 
-      expect($ctrl.cartData).toEqual('some data')
+      expect($ctrl.cartData).toEqual(cartData)
       expect($ctrl.errorLoadingCart).toEqual(false)
+      expect($ctrl.brandedAnalyticsFactory.addPaymentInfo).toHaveBeenCalled(cartData.items[0])
```
Comment (dr-bizz) [inline src/app/branded/step-2/branded-checkout-step-2.spec.js:45]: What about doing a test with the cover fee?

Comment (canac) [reply src/app/branded/step-2/branded-checkout-step-2.spec.js]: Good catch. `toHaveBeenCalled` should have been `toHaveBeenCalledWith` also.

**Review by dr-bizz — APPROVED** [review]

**Pull request merged**